### PR TITLE
Run upstream e2e test with net-istio

### PR DESCRIPTION
**Pull request by nak3**
/cc

### CODE REVIEW

**Comment by nak3** [comment]
/retest
istio-operator [did not start](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1402821148621672448/artifacts/upstream-e2e-aws-ocp-47/gather-extra/artifacts/pods/openshift-operators_istio-operator-bb69b78d4-zt4qw_istio-operator_previous.log).

**Comment by nak3** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1402865094089510912

Websocket test could be fixed by https://github.com/openshift/knative-serving/pull/802
DomainMapping and gRPC tests do not work as expected.
`TestContainerExitingMsg` and `TestImagePullError`  are strange :thinking:

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra issue.

**Comment by nak3** [comment]
domainmapping test hits this issue https://github.com/knative/serving/commit/26c93b9c4af345ef4231659c4c0ac75ef35bec71  

```
    stream.go:*54: E 07:41:55.348 domain-mapping-744987c5fb-zl6xs [knative.dev.serving.pkg.reconciler.domainmapping.Reconciler] [serving-tests/domain-mapping-bgboydtt.apps.ci-op-mcifrq*q-fbf8c.origin-ci-int-aws.dev.rhcloud.com] Returned an error err=failed to create Ingress: Ingress.networking.internal.knative.dev "domain-mapping-bgboydtt.apps.ciefb90bedf33484b06ac7d3ba94496ca7" is invalid: metadata.labels: Invalid value: "domain-mapping-bgboydtt.apps.ci-op-mcifrq*q-fbf8c.origin-ci-int-aws.dev.rhcloud.com": must be no more than 63 characters
```

**Comment by nak3** [comment]
`TestCustomOpenShiftRoute` fails because it has fixed `kourier` and `knative-serving-ingress` values.

**Comment by nak3** [comment]
/retest

infra

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47
/test 4.7-operator-e2e-aws-ocp-47

failed to create cluster.

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47
/test 4.7-upgrade-tests-aws-ocp-47

failed to create cluster again.

**Comment by nak3** [comment]
/retest

VPC issue.

**Comment by nak3** [comment]
/test 4.7-upgrade-tests-aws-ocp-47

infra issue.

**Comment by nak3** [comment]
grpc_test.go must be removed. It hangs and got timeout.

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra.

**Comment by nak3** [comment]
/retest

infra...

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra.

**Comment by nak3** [comment]
/test 4.7-upgrade-tests-aws-ocp-47

Eventing's upgrade error.

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra.

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra.

**Comment by nak3** [comment]
/test 4.7-upstream-e2e-aws-ocp-47

infra.

**Comment by nak3** [comment]
Aletmanager prints warning https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1407509003751133184/artifacts/upstream-e2e-aws-ocp-47/serverless-e2e/artifacts/build-TQsn6lF/alerts.json

Hmm... it should be fixed by https://github.com/openshift-knative/serverless-operator/pull/1036

**Comment by nak3** [comment]
https://github.com/openshift-knative/serverless-operator/pull/1036 fixed the issue, but it sets to `none` only when no setting. (It is correct fix.)
The original CR`test/v1alpha1/resources/operator.knative.dev_v1alpha1_knativeserving_cr.yaml` sets to `prometheus` so we need to override.

**Comment by nak3** [comment]
The test passed. Now trying to remove `GODEBUG="x509ignoreCN=0"`.

**Comment by nak3** [comment]
```
    domain_mapping_test.go:110: CheckDistribution=domain domain-mapping-cvtnhyal.apps.ci-op-kzsmj52w-fbf8c.origin-ci-int-aws.dev.rhcloud.com failed: want at least 200, got 0 for response "What a spaceport!",domain domain-mapping-cvtnhyal.apps.ci-op-kzsmj52w-fbf8c.origin-ci-int-aws.dev.rhcloud.com: saw expected responses 0 times, wanted 200, expected no error
```

This backport https://github.com/openshift/knative-serving/pull/805 was not used yet.

**Comment by nak3** [comment]
TestDomainMapping passed but TestDomainMapping [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1407995861476052992).
It just needs to scale up domainmapping deployment before the test.
The current code did not add `--enable-alphe` so it missed the test.

**Comment by nak3** [comment]
/retest

Hmmm... `TestDomainMapping` passed before but it failed now.
Why `RetryingRouteInconsistency` did not retry.

```
    spoof.go:172: Retrying http://domain-mapping-dqqjwkng.apps.ci-op-cp681k6x-fbf8c.origin-ci-int-aws.dev.rhcloud.com: retrying for: Get "https://domain-mapping-dqqjwkng.apps.ci-op-cp681k6x-fbf8c.origin-ci-int-aws.dev.rhcloud.com/": EOF
    domain_mapping_test.go:220: Error probing http://domain-mapping-dqqjwkng.apps.ci-op-cp681k6x-fbf8c.origin-ci-int-aws.dev.rhcloud.com: response: status: 503, body: <html>^M
          <head>^M
            <meta name="viewport" content="width=device-width, initial-scale=1">^M
        ^M
          <style type="text/css">^M
          /*!^M
           * Bootstrap v3.3.5 (http://getbootstrap.com)^M
```

**Comment by nak3** [comment]
/retest

infra (yum) error.

**Comment by nak3** [comment]
/retest

New failure with `TestActivatorHAGraceful `. I think it is flake.

**Comment by nak3** [comment]
`TestActivatorHAGraceful` failed again due to `Failed to delete pod activator-85d6cdc89-dr8bg: pods "activator-85d6cdc89-dr8bg" not found`.

It will be improved by https://github.com/knative/serving/commit/9054d8b036f34d0127e06661d32a8c8437b18ff8 :thinking:

**Comment by nak3** [comment]
/retest

`TestActivatorHAGraceful` works on my local even with openshift/release-v0.22.0 branch.

**Comment by nak3** [comment]
`TestActivatorHANonGraceful` also has same problem :disappointed_relieved:  
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1409337331164385280

**Comment by nak3** [comment]
/retest

Test passed once. One more time.

**Comment by nak3** [comment]
Rebased as `/retest` did not trigger the test.

**Comment by nak3** [comment]
Serving 0.22 didn't have h2c issue but 0.23 got the issue [log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1410082954947858432):

```
    http2_test.go:67: The endpoint http://hello-http2-with-por-fwbdxyth-serving-tests.apps.ci-op-17k619p9-fbf8c.origin-ci-int-aws.dev.rhcloud.com for Route hello-http2-with-por-fwbdxyth didn't serve the expected text "Hello, New World! How about donuts and coffee?": response: status: 426, body: , headers: map[Content-Length:[0] Date:[Wed, 30 Jun 2021 04:54:46 GMT] Server:[istio-envoy] X-Envoy-Upstream-Service-Time:[2] Zipkin_trace_id:[d47245bee119f7711504b28250809ca0]] did not pass checks: status = 426 426 Upgrade Required, want one of: [200]
```
I experienced it in https://github.com/openshift/knative-serving/pull/751 but still not sure why only serving 0.23 happens this problem.

**Comment by nak3** [comment]
/retest

TestRouteCreation failed. (It is the first time to fail it.)
```
    route_test.go:158: CheckSLO() error SLI for "TestRouteCreation" = 0.915663, wanted >= 1.000000
```
[log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1014/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1410228017346646016)

**Comment by nak3** [comment]
Only two HA tests failed.
`TestDomainMappingHA` ... patch command overrides domainmapping's HA so needs to fix script.
` TestControllerHA` ... it needs https://github.com/openshift/knative-serving/pull/836

**Comment by nak3** [comment]
Umm... TestControllerHA still failed. https://github.com/openshift/knative-serving/pull/836 might not be applied.

**Comment by nak3** [comment]
/cc @markusthoemmes @mgencur @maschmid 

This is ready for review. Could you please take a look?

**Comment by skonto** [comment]
@nak3 should we run downstream tests with this setup as well? Is this coming next?

**Comment by nak3** [comment]
Yes, I am going to add the mesh test for downstream with follow up PR. (Because it messes up script and hard to review I guess 
 :sweat_smile: )

File: Makefile
Comment (markusthoemmes) [inline Makefile:84]: Should this go away before a merge?

File: test/serving.bash
Comment (markusthoemmes) [inline test/serving.bash:64]: This means that we'll have to document gRPC/HTTP2 not working for mesh for now, right?

File: Makefile
Comment (nak3) [reply Makefile:84]: Yes, correct. This is a temporary as pre-submit PR always run `test-upstream-e2e-no-upgrade`. I will revert it and trigger job against `test-upstream-e2e-mesh`.

File: test/serving.bash
Comment (nak3) [reply test/serving.bash:64]: Yes, I will open the JIRA.

File: hack/lib/serverless.bash
Comment (markusthoemmes) [inline hack/lib/serverless.bash:191]: Is this required? Don't we do that "automatically"?

File: hack/lib/serverless.bash
Comment (markusthoemmes) [inline hack/lib/serverless.bash:188]: Why do we need this? Don't we also need that in "usual" setups?

File: test/serving.bash
Comment (markusthoemmes) [inline test/serving.bash:58]: We should be able to fix that. Can we update the `openssl` command in https://github.com/openshift-knative/serverless-operator/blob/main/openshift/ci-operator/build-image/Dockerfile?

Comment (nak3) [reply test/serving.bash:58]: I tried but the base image of `registry.ci.openshift.org/openshift/release:golang-1.15` is CentOS 7 and the latest openssl command was still 1.10.

```
$ docker run build-image rpm -qf `which openssl`
openssl-1.0.2k-21.el7_9.x86_64
```

File: hack/lib/serverless.bash
Comment (nak3) [reply hack/lib/serverless.bash:188]: Yes, we have the `replicas: 2` in "usual" setups.
https://github.com/openshift-knative/serverless-operator/blob/b368a3fe89f5c39c3582190f04ead737265c0ecf/test/v1alpha1/resources/operator.knative.dev_v1alpha1_knativeserving_cr.yaml#L7-L8
But the `spec.deployments` is an array. So when we patch the annotation (inject and rewriteAppHTTPProbers) to `spec.deployments` for activator and autoscaler, `domain-mapping` is dropped :sweat:

Comment (markusthoemmes) [reply hack/lib/serverless.bash:188]: Gotcha, makes sense!

File: hack/lib/serverless.bash
Comment (nak3) [reply hack/lib/serverless.bash:191]: Umm... we have the default as `"prometheus"` 
https://github.com/openshift-knative/serverless-operator/blob/b368a3fe89f5c39c3582190f04ead737265c0ecf/test/v1alpha1/resources/operator.knative.dev_v1alpha1_knativeserving_cr.yaml#L39
so I think we need to override it with `none`. Sorry if I misunderstood the meaning of "automatically".

Comment (markusthoemmes) [reply hack/lib/serverless.bash:191]: ahhhh we explicitly set it. Nevermind then, I was thinking in our Golang code, where we default to `none` currently, if istio is enabled.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

/hold

Please revert the "old" make target and then we're good to go here :)

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
